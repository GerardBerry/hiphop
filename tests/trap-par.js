"use hopscript"

var rjs = require("../lib/reactive-js.js");

var sigA = new rjs.Signal("A");
var sigB = new rjs.Signal("B");
var sigC = new rjs.Signal("C");

var prg = <rjs.reactivemachine debug name="trappar">
  <rjs.outputsignal ref=${sigA}/>
  <rjs.outputsignal ref=${sigB}/>
  <rjs.outputsignal ref=${sigC}/>
  <rjs.trap trap_name="T">
    <rjs.parallel>
      <rjs.sequence>
	<rjs.emit signal_name="A"/>
	<rjs.exit trap_name="T"/>
      </rjs.sequence>
      <rjs.sequence>
	<rjs.emit signal_name="B"/>
	<rjs.pause/>
	<rjs.emit signal_name="C"/>
      </rjs.sequence>
    </rjs.parallel>
  </rjs.trap>
</rjs.reactivemachine>;

exports.prg = prg
