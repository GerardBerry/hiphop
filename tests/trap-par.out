trappar> ;
--- Output: A B
trappar> ;
--- Output:
trappar> ;
--- Output:
trappar> ;
--- Output:
trappar> !reset;
--- Automaton trappar reset
trappar> ;
--- Output: A B
trappar> ;
--- Output:
